homeassistant/components/accuweather/sensor.py

```diff
@@ -87,72 +109,58 @@ def device_info(self):
-                FORECAST_SENSOR_TYPES[self.kind][ATTR_DEVICE_CLASS]
-                == DEVICE_CLASS_TEMPERATURE
-            ):
-                return self.coordinator.data[ATTR_FORECAST][self.forecast_day][
-                    self.kind
-                ]["Value"]
-            if self.kind in ["WindDay", "WindNight", "WindGustDay", "WindGustNight"]:
-                return self.coordinator.data[ATTR_FORECAST][self.forecast_day][
-                    self.kind
-                ]["Speed"]["Value"]
-            if self.kind in ["Grass", "Mold", "Ragweed", "Tree", "UVIndex", "Ozone"]:
-                return self.coordinator.data[ATTR_FORECAST][self.forecast_day][
-                    self.kind
-                ]["Value"]
-            return self.coordinator.data[ATTR_FORECAST][self.forecast_day][self.kind]
+            if self._description["device_class"] == DEVICE_CLASS_TEMPERATURE:
+                return cast(float, self._sensor_data["Value"])
+            if self.kind == "UVIndex":
+                return cast(int, self._sensor_data["Value"])
+        if self.kind in ["Grass", "Mold", "Ragweed", "Tree", "Ozone"]:
+            return cast(int, self._sensor_data["Value"])
         if self.kind == "Ceiling":
-            return round(self.coordinator.data[self.kind][self._unit_system]["Value"])
+            return round(self._sensor_data[self._unit_system]["Value"])
         if self.kind == "PressureTendency":
-            return self.coordinator.data[self.kind]["LocalizedText"].lower()
-        if SENSOR_TYPES[self.kind][ATTR_DEVICE_CLASS] == DEVICE_CLASS_TEMPERATURE:
-            return self.coordinator.data[self.kind][self._unit_system]["Value"]
+            return cast(str, self._sensor_data["LocalizedText"].lower())
+        if self._description["device_class"] == DEVICE_CLASS_TEMPERATURE:
+            return cast(float, self._sensor_data[self._unit_system]["Value"])
         if self.kind == "Precipitation":
-            return self.coordinator.data["PrecipitationSummary"][self.kind][
-                self._unit_system
-            ]["Value"]
+            return cast(float, self._sensor_data[self._unit_system]["Value"])
         if self.kind in ["Wind", "WindGust"]:
-            return self.coordinator.data[self.kind]["Speed"][self._unit_system]["Value"]
-        return self.coordinator.data[self.kind]
+            return cast(float, self._sensor_data["Speed"][self._unit_system]["Value"])
+        if self.kind in ["WindDay", "WindNight", "WindGustDay", "WindGustNight"]:
+            return cast(StateType, self._sensor_data["Speed"]["Value"])
+        return cast(StateType, self._sensor_data)
 
     @property
-    def icon(self):
+    def icon(self) -> str | None:
         """Return the icon."""
         if self.forecast_day is not None:
-            return FORECAST_SENSOR_TYPES[self.kind][ATTR_ICON]
-        return SENSOR_TYPES[self.kind][ATTR_ICON]
+            return self._description["icon"]
+        return self._description["icon"]
```

Comment:
Remove the condition 🙂
```suggestion
        return self._description["icon"]
```

Reply:
Oh my, I'm blind?! 🤦‍♂️